### PR TITLE
Set loaded_from before using extension_dir to ensure it is correct for --user or other alternate install path modes

### PR DESCRIPTION
**Pull request by coderanger**
Fixes https://github.com/rubygems/rubygems/issues/1680

Not 100% sure the best way to write a test for this.

I will abide by the [code of conduct](https://github.com/rubygems/rubygems/blob/master/CODE_OF_CONDUCT.md).

### CODE REVIEW

**Comment by coderanger** [comment]
Now with an actual test.

**Comment by coderanger** [comment]
On further checking, I don't think my test is good enough to catch a regression of this.

**Comment by coderanger** [comment]
Amended the test again, this time I think it's 👍

File: lib/rubygems/installer.rb [hunk 1]
```diff
@@ -282,18 +282,23 @@ def install
...
+
     # Completely remove any previous gem files
     FileUtils.rm_rf gem_dir
     FileUtils.rm_rf spec.extension_dir
 
     FileUtils.mkdir_p gem_dir
 
-    if @options[:install_as_default]
-      spec.loaded_from = default_spec_file
+    if @options[:install_as_default] then
```
Comment (godfat) [inline lib/rubygems/installer.rb]: Do we need to use `then` here?

**Comment by godfat** [comment]
Thank you for the patch and tests!

File: lib/rubygems/installer.rb (hunk 1 above)
Comment (coderanger) [inline lib/rubygems/installer.rb]: It's the official coding style according to `CONTRIBUTING.md`, I defer to others as to if that is accurate.

Comment (godfat) [inline lib/rubygems/installer.rb]: Apparently I am not reading it O_o Sorry about that.

**Comment by segiddins** [comment]
Excellent! 👍

**Comment by segiddins** [comment]
@homu r+

**Pull request merged**